test/libsolidity/semanticTests/externalContracts/ramanujan_pi.sol

```diff
@@ -0,0 +1,45 @@
+==== ExternalSource: _prbmath/PRBMathCommon.sol ====
+==== ExternalSource: _prbmath/PRBMathSD59x18.sol ====
+==== Source: prbmath.sol ====
+import "_prbmath/PRBMathSD59x18.sol";
+
+// The goal of this test file is to implement Ramanujan's pi approximation using various libraries.
+
+function factorial(uint n) pure returns (uint ret) {
+    ret = 1;
+    for (; n > 1; --n)
+        ret *= n;
+}
+
+contract test {
+    using PRBMathSD59x18 for int256;
+    
+    function prb_scale(uint n) internal pure returns (int256 ret) {
+        // Scale to SD59x18
+        ret = int256(n * 10e17);
+    }
+
+    // This dumb implementation of Ramanujan series calculates 1/pi
+    function prb_pi() external pure returns (int256 ret) {
```

Comment:
Based on (where it is much more readable): https://github.com/wasmx/fizzy/blob/master/test/benchmarks/ramanujan_pi.c

Reply:
It should be possible to write this as the following using native fixed and the stdlib:
```
        uint n = 6; // Number of iterations
        fixed ret;
        // We assume factorial(fixed) -> fixed exists as a helper above, and pow/sqrt are part of the stdlib.
        for (fixed k = 0; k < fixed(n); k++) {
            fixed a = factorial(4.0 * k) / pow(factorial(k), 4.0);
            fixed b = (25390.0 * k + 1103.0) / pow(396.0, 4.0 * k);
            ret += a * b;
        }
        ret *= (2.0 * sqrt(2.0)) / pow(99.0, 2.0); 
        ret = 1.0 / ret;
```

Reply:
`pow(..., 4.0)`, i.e., fixed point as exponent is probably too much for stdlib.

Reply:
Sure, I thought it would do an int conversion internally and felt lazy to write it out, or to write it as `pow(uint(factorial(k)), 4)`.

Reply:
Ok here's another version then:
```
        uint n = 6; // Number of iterations
        fixed ret;
        // We assume factorial(uint) -> uint exists as a helper above, and pow/sqrt are part of the stdlib.
        for (uint k = 0; k < n; k++) {
            fixed a = fixed(factorial(4 * k)) / pow(fixed(factorial(k)), 4);
            fixed b = (25390.0 * fixed(k) + 1103.0) / pow(396.0, 4 * k);
            ret += a * b;
        }
        ret *= (2.0 * sqrt(2.0)) / pow(99.0, 2.0); 
        ret = 1.0 / ret;
```